vtex/utils/fetchVTEX.ts

```diff
@@ -60,6 +60,15 @@ const getSanitizedInput = (
     }
   });
 
+  const productNameContains = url.searchParams.get("productNameContains");
+
+  if (productNameContains) {
+    return url.toString().replace(
+      /productNameContains=([^&#]*)/,
+      `productNameContains=${encodeURIComponent(productNameContains)}`,
+    );
+  }
+
```

Comment:
```suggestion

  if (productNameContains) {
    url.searchParams.set("productNameContains",  encodeURIComponent(productNameContains))
  }

```

What about? (test before plz)

Reply:
That's how it worked too, but in this case I changed it to encode directly in the loader, it still works, is that better?